src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
Change the code to avoid repetition, but save readability.
Don’t shorten variables or function names. Check this [guide](https://mate-academy.github.io/style-guides/javascript.html#2311-dont-shorten-variables-or-functions-names).
Replace ```find``` with ```some``` because it returns more suitable boolean value.
```suggestion
    const women = people.filter((person) => (
      withChildren
        ? person.sex === 'f'
        : person.sex === 'f' && people.some(child => child.mother === person.name)
      )
    );
```